openvdb/openvdb/python/CMakeLists.txt

```diff
@@ -138,63 +62,46 @@ set(OPENVDB_PYTHON_MODULE_SOURCE_FILES
   pyVec3Grid.cc
 )
 
-if(NOT DEFINED PYOPENVDB_INSTALL_DIRECTORY)
-  get_filename_component(Python_PACKAGES_DIR ${Python_SITELIB} NAME)
-  set(PYOPENVDB_INSTALL_DIRECTORY
-    ${CMAKE_INSTALL_LIBDIR}/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/${Python_PACKAGES_DIR}
-    CACHE STRING "The directory to install the pyopenvdb.so module."
-  )
-endif()
-
-pybind11_add_module(pyopenvdb ${OPENVDB_PYTHON_MODULE_SOURCE_FILES})
-
-target_link_libraries(pyopenvdb PUBLIC
-  ${OPENVDB_LIB}
-  ${OPENVDB_PYTHON_DEPS}
-)
-
+target_link_libraries(openvdb_python PUBLIC ${OPENVDB_LIB})
 if(OPENVDB_PYTHON_WRAP_ALL_GRID_TYPES)
-  target_compile_definitions(pyopenvdb PRIVATE "-DPY_OPENVDB_WRAP_ALL_GRID_TYPES")
+  target_compile_definitions(openvdb_python PRIVATE "-DPY_OPENVDB_WRAP_ALL_GRID_TYPES")
 endif()
 if(USE_NUMPY)
-  target_compile_definitions(pyopenvdb PUBLIC "-DPY_OPENVDB_USE_NUMPY")
+  target_compile_definitions(openvdb_python PUBLIC "-DPY_OPENVDB_USE_NUMPY")
 endif()
 if(USE_AX)
-  target_link_libraries(pyopenvdb PUBLIC ${OPENVDB_AX_LIB})
-  target_compile_definitions(pyopenvdb PUBLIC "-DPY_OPENVDB_USE_AX")
+  target_link_libraries(openvdb_python PUBLIC ${OPENVDB_AX_LIB})
+  target_compile_definitions(openvdb_python PUBLIC "-DPY_OPENVDB_USE_AX")
 endif()
+set_target_properties(openvdb_python PROPERTIES OUTPUT_NAME "openvdb")
```

Comment:
The resulting library name is `openvdb.cpython-310-linux-x86_64.so`. If this library is directly included in the site-packages directory as it is currently, then the name of the module (i.e. `openvdb`) needs to match the module name specified in import (i.e. `import openvdb`). Otherwise, Python won't find the module.

My follow-up pip install PR will decouple this somewhat. The general plan is to rename the .so to `_openvdb.cpython-310-linux-x86_64.so` with the underscore indicating that the C++ implementation is intended to be hidden (a generally accepted convention in Python packages). Then, there will be a lightweight wrapper module called `openvdb` that imports the C++ module members into the `openvdb` namespace. However, this approach requires more file shuffling that without pip would just be additional burden on the user hence it's exclusion from this PR.